treeherder/webapp/api/bug.py

```diff
@@ -11,20 +11,19 @@
 class BugJobMapViewSet(viewsets.ViewSet):
 
     def create(self, request, project):
-        """
-        Add a new relation between a job and a bug
-        """
-        job_id, bug_id = map(int, (request.data['job_id'],
-                                   request.data['bug_id']))
+        """Add a new relation between a job and a bug."""
+        job_id = int(request.data['job_id'])
+        bug_id = int(request.data['bug_id'])
+
+        bug_map = BugJobMap.objects.filter(job_id=job_id, bug_id=bug_id).first()
+        if bug_map:
+            return Response({"message": "Bug job map skipped: mapping already exists"}, 200)
```

Comment:
What do you think about a try-except around a `.create()` to reduce the number of queries generated in the common case (no mapping exists already) from 2 to 1?

Reply:
Yep, good point! Fixed in 5df694c77